README.md

```diff
@@ -54,6 +54,28 @@ Then, inside your own project, use `cargo +nightly miri` to run your project, if
 a bin project, or run `cargo +nightly miri test` to run all tests in your project
 through miri.
 
+## Running miri with full libstd
+
+Per default libstd does not contain the MIR of non-polymorphic functions.  When
+miri hits a call to such a function, execution terminates.  To fix this, it is
+possible to compile libstd with full MIR:
+
+```sh
+cargo install xargo
+cd xargo/
+RUSTFLAGS='-Zalways-encode-mir' xargo build --target `rustc -vV | egrep '^host: ' | sed 's/^host: //'`
+```
+
+Now you can run miri against the libstd compiled by xargo:
+
+```sh
+cargo run --bin miri -- --sysroot ~/.xargo/HOST tests/run-pass/vecs.rs
+```
+
+Notice that you will have to re-run the last step of the preparations above when
+your toolchain changes (e.g., when you update the nightly).  Also, xargo doesn't
+currently work with nightlies newer than 2017-04-23.
```

Comment:
I used the old miri from back then.
I'm currently trying to fix the creation of the rust-src distribution component such that xargo works again.